cdap-master/src/main/java/co/cask/cdap/data/tools/HBaseQueueDebugger.java

```diff
@@ -303,28 +405,123 @@ public long getTotal() {
       return unprocessed + processedAndVisible + processedAndNotVisible;
     }
 
-    public String getReport() {
-      return String.format("unprocessed: %d; processed and visible: %d; processed and not visible: %d; total: %d",
-                           getUnprocessed(), getProcessedAndVisible(), getProcessedAndNotVisible(), getTotal());
+    public Optional<Long> getMinWritePointer() {
+      return minWritePointer;
+    }
+
+    public String getMinWritePointerString() {
+      if (minWritePointer.isPresent()) {
+        return Long.toString(minWritePointer.get());
+      } else {
+        return "n/a";
+      }
+    }
+
+    public String getMinWritePointerTimestampString() {
+      if (minWritePointer.isPresent()) {
+        return Long.toString(minWritePointer.get() / TxConstants.MAX_TX_PER_MS);
+      } else {
+        return "n/a";
+      }
+    }
+
+    private String getTxTimestampReport() {
+      return String.format("min tx timestamp: %s", getMinWritePointerTimestampString());
+    }
+
+    private String getDetailedReport() {
+      return String.format("min write pointer: %s; unprocessed: %d; processed and visible: %d; " +
+                             "processed and not visible: %d; total: %d",
+                           getMinWritePointerString(), getUnprocessed(), getProcessedAndVisible(),
+                           getProcessedAndNotVisible(), getTotal());
+    }
+
+    public String getReport(boolean showTxTimestampOnly) {
+      if (showTxTimestampOnly) {
+        return getTxTimestampReport();
+      } else {
+        return getDetailedReport();
+      }
+    }
+
+    public void add(QueueStatistics stats) {
+      if (stats.getMinWritePointer().isPresent()) {
+        recordMinWritePointer(stats.getMinWritePointer().get());
+      }
+      countUnprocessed(stats.getUnprocessed());
+      countProcessedAndNotVisible(stats.getProcessedAndNotVisible());
+      countProcessedAndVisible(stats.getProcessedAndVisible());
     }
   }
 
+  public static HBaseQueueDebugger createDebugger() {
+    Injector injector = Guice.createInjector(
+      new ConfigModule(CConfiguration.create(), HBaseConfiguration.create()),
+      new IOModule(),
+      new ZKClientModule(),
+      new LocationRuntimeModule().getDistributedModules(),
+      new DiscoveryRuntimeModule().getDistributedModules(),
+      new StreamAdminModules().getDistributedModules(),
+      new NotificationFeedClientModule(),
+      new TwillModule(),
+      new ExploreClientModule(),
```

Comment:
`HBaseQueueDebugger` depends on `Store` which appears to require these guice modules.